bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/discovery/TouchWandControllerDiscoveryService.java

```diff
@@ -0,0 +1,145 @@
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.touchwand.internal.TouchWandBindingConstants;
+import org.openhab.binding.touchwand.internal.TouchWandBridgeHandler;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TouchWandControllerDiscoveryService} Discovery service for Touchwand Controllers.
+ *
+ * @author Roie Geron - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = false, configurationPid = "discovery.touchwand")
+@NonNullByDefault({})
+public class TouchWandControllerDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int SEARCH_TIME = 2;
+    private static final int TOUCHWAND_BCAST_PORT = 35000;
+    private Thread socketReceiveThread = null;
+    private DatagramSocket listenSocket = null;
+
+    private final Logger logger = LoggerFactory.getLogger(TouchWandControllerDiscoveryService.class);
+
+    public TouchWandControllerDiscoveryService() {
+        super(TouchWandBridgeHandler.SUPPORTED_THING_TYPES, SEARCH_TIME, true);
+        removeOlderResults(getTimestampOfLastScan());
+    }
+
+    @Override
+    protected void startScan() {
+        try {
+            listenSocket = new DatagramSocket(TOUCHWAND_BCAST_PORT);
+        } catch (SocketException e) {
+            logger.warn("SocketException {}", e.getMessage());
+        }
+        runReceiveThread(listenSocket);
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+        if (socketReceiveThread != null) {
+            socketReceiveThread.interrupt();
+            socketReceiveThread = null;
+        }
+        if (listenSocket != null) {
+            listenSocket.close();
+        }
+    }
+
+    private void addDeviceDiscoveryResult(String label, String ip) {
+        String id = ip.replaceAll("\\.", "");
+        ThingUID thingUID = new ThingUID(THING_TYPE_BRIDGE, id);
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("label", label);
+        properties.put("ipAddress", ip);
+        // @formatter:off
+        logger.debug("Add new Bridge label:{} id {} ",label, id);
+        thingDiscovered(DiscoveryResultBuilder.create(thingUID)
+                .withThingType(THING_TYPE_BRIDGE)
+                .withLabel(label)
+                .withProperties(properties)
+                .withRepresentationProperty(ip)

```

Comment:
The argument must be the *name* of the property.
```suggestion
                .withRepresentationProperty("ipAddress")

```